subgraphs/rocket-pool/src/updaters/financialMetrics.ts

```diff
@@ -0,0 +1,461 @@
+    Address.fromString(RPL_ADDRESS),
+    block.number
+  ).lastPriceUSD!;
+
+  log.error("[updateTotalRewardsMetrics] additional Rewards: {}", [
+    additionalRewards.toString(),
+  ]);
+
+  const newDailyEmissions =
+    poolMetricsDailySnapshot.rewardTokenEmissionsAmount![0].plus(
+      additionalRewards
+    );
+  const newHourlyEmissions =
+    poolMetricsDailySnapshot.rewardTokenEmissionsAmount![0].plus(
+      additionalRewards
+    );
+
+  log.error("[updateTotalRewardsMetrics] lastRewardPriceUsd: {}", [
+    lastRewardPriceUsd.toString(),
+  ]);
+  log.error("[updateTotalRewardsMetrics] newDailyEmissions: {}", [
+    newDailyEmissions.toString(),
+  ]);
+  log.error("[updateTotalRewardsMetrics] newHourlyEmissions: {}", [
+    newHourlyEmissions.toString(),
+  ]);
+
+  poolMetricsDailySnapshot.rewardTokenEmissionsAmount = [newDailyEmissions];
+
+  poolMetricsDailySnapshot.rewardTokenEmissionsUSD = [
+    bigIntToBigDecimal(newDailyEmissions).times(lastRewardPriceUsd),
+  ];
+
+  poolMetricsDailySnapshot.save();
+
+  log.error(
+    "[updateTotalRewardsMetrics] poolMetrics reward token emissions: {}",
+    [poolMetricsDailySnapshot.rewardTokenEmissionsUSD![0].toString()]
+  );
+
+  poolMetricsHourlySnapshot.rewardTokenEmissionsAmount! = [newHourlyEmissions];
+  poolMetricsHourlySnapshot.rewardTokenEmissionsUSD! = [
+    bigIntToBigDecimal(newHourlyEmissions).times(lastRewardPriceUsd),
+  ];
+  poolMetricsHourlySnapshot.save();
+
+  financialMetrics.save();
+}
+
+export function updateTotalRevenueMetrics(
+  block: ethereum.Block,
+  additionalRevenue: BigInt,
+  totalShares: BigInt // of rETH
+): void {
+  const pool = getOrCreatePool(block.number, block.timestamp);
+  const protocol = getOrCreateProtocol();
+  const financialMetrics = getOrCreateFinancialDailyMetrics(block);
+  const poolMetricsDailySnapshot = getOrCreatePoolsDailySnapshot(block);
+  const poolMetricsHourlySnapshot = getOrCreatePoolsHourlySnapshot(block);
+
+  const additionalRewards = bigIntToBigDecimal(additionalRevenue);
+
+  const lastPriceUsd = getOrCreateToken(
+    Address.fromString(ETH_ADDRESS),
+    block.number
+  ).lastPriceUSD!;
+
+  // Pool
+  pool.cumulativeTotalRevenueUSD = pool.cumulativeTotalRevenueUSD.plus(
+    additionalRewards.times(lastPriceUsd)
+  );
+
+  pool.outputTokenSupply = totalShares;
+
+  pool.outputTokenPriceUSD = getOrCreateToken(
+    Address.fromString(PROTOCOL_ID),
+    block.number
+  ).lastPriceUSD;
+  pool.save();
+
+  // Pool Daily
+  poolMetricsDailySnapshot.cumulativeTotalRevenueUSD =
+    pool.cumulativeTotalRevenueUSD;
+  poolMetricsDailySnapshot.dailyTotalRevenueUSD =
+    poolMetricsDailySnapshot.dailyTotalRevenueUSD.plus(
+      additionalRewards.times(lastPriceUsd)
+    );
+  poolMetricsDailySnapshot.outputTokenSupply = pool.outputTokenSupply;
+  poolMetricsDailySnapshot.outputTokenPriceUSD = pool.outputTokenPriceUSD;
+  poolMetricsDailySnapshot.stakedOutputTokenAmount = pool.outputTokenSupply;
```

Comment:
should this be pool.stakedOutputTokenAmount?

Reply:
And should `pool.stakedOutputTokenAmount` be updated anywhere? I don't see it being set ever. The days where we only create the market snapshots are the days where it is set to 0 bc the outputTokenAmount is never updated

Reply:
annnnnd thats the bug I missed